arch/x86/include/asm/kvm_pkvm.h

```diff
@@ -60,23 +60,82 @@ enum pkvm_hc {
 	MAX_PKVM_HYPERCALLS,
 };
 
+#define PKVM_HC_DATA_MAX_NUM		4
+
+union pkvm_hc_data {
+	struct {
+		u64 data[PKVM_HC_DATA_MAX_NUM];
+	} raw;
+};
+
+/*
+ * The union pkvm_hc_data is passed in hypercalls via the registers used for
+ * hypercall arguments, and PKVM_HC_DATA_MAX_NUM represents the number of those
+ * registers. So the size of the union cannot be larger than that.
+ */
+static_assert(sizeof(union pkvm_hc_data) == PKVM_HC_DATA_MAX_NUM * sizeof(u64));
+
+#define PKVM_HC_DATA_NUM(f)		\
+	(ALIGN(sizeof(((union pkvm_hc_data *)0)->f), sizeof(u64)) / sizeof(u64))
+
+#define PKVM_HC_OUTPUT_NUM(f)	f##_output_num
+
+enum pkvm_hc_output_data_num {
```

Comment:
nit: looks like this enum can be anonymous, we don't need a name for it?

And same about `enum pkvm_hc_input_data_num`.

So it all can be reduced to this?

```
enum {
	#define PKVM_HC(f)	PKVM_HC_OUTPUT_NUM(f) = 0,
	#define PKVM_HC_OUT(f)	PKVM_HC_OUTPUT_NUM(f) = PKVM_HC_DATA_NUM(f),
	#include <asm/pkvm_hypercalls.h>

	#define PKVM_HC(f)	PKVM_HC_INPUT_NUM(f) = 0,
	#define PKVM_HC_IN(f)	PKVM_HC_INPUT_NUM(f) = PKVM_HC_DATA_NUM(f),
	#include <asm/pkvm_hypercalls.h>
};
```